requirements/dev.txt

```diff
@@ -7,3 +7,4 @@ pytest
 pytest-annotate
 tox
 types-setuptools
+neo4j
```

Comment:
We don't need it here since it's already present in `requirements/base.txt`. Please see this link https://github.com/neo4j/graph-data-science-client/blob/main/CONTRIBUTING.md#specifically-for-this-project for how to set up dev env for the project :)